### ISSUE TITLE
v0.8.0 representation glitches

### ISSUE THREAD

**Issue by brunopostle** [IfcOpenShell/IfcOpenShell#4848]
This model [_courtyard.ifc.txt](https://github.com/user-attachments/files/15799625/_courtyard.ifc.txt) opens ok in v0.7.0, but in v.0.8.0 BlenderBIM doesn't show parapets (drawn with `IfcSurfaceCurveSweptAreaSolid`) and the roof pitch (drawn with a `IfcBooleanResult UNION`) has failed.

![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/1747315/020da1ee-6525-43a4-b2d2-0738090849a3)

IfcConvert shows this error:

    Log:
    [Error] [2024-06-12 07:52:47] {0ylklUYwjRjHHAZ0ch23qQ} Not implemented
    #19520=IfcSurfaceCurveSweptAreaSolid(#19504,#19518,#19516,0.,1.,#19519)
    [Error] [2024-06-12 07:52:47] {0MYWhUWy0tQyeKjVQ98dwA} Not implemented
    #19553=IfcSurfaceCurveSweptAreaSolid(#19537,#19551,#19549,0.,1.,#19552)

(In my other homemaker project I stopped using `IfcSurfaceCurveSweptAreaSolid`, since they are not editable in BlenderBIM anyway, maybe I'll switch here too)

#4832

**Comment by brunopostle** [comment]
The IfcBooleanResult UNION glitch seems to be fixed in a new build. I can cope with `IfcSurfaceCurveSweptAreaSolid` being unsupported if necessary.

**Comment by aothms** [comment]
No, they will come back, it's not a  fundamental problem, it's just that the face parametrization (also in advanced brep) still needs some work. But since this is rather specific and isolated I'll give priority first to the more pressing issues.

**Comment by aothms** [comment]
I made *some* progress, but it looks still rather broken. Need to have a bit better look at how all the matrices line up in the newly introduced taxonomy. That is still a bit of an inconsistent area.

**Comment by Moult** [comment]
Just checked with the latest state of things, still segfaulting.

**Comment by brunopostle** [comment]
This test file now doesn't segfault here, and the `IfcSurfaceCurveSweptAreaSolid` representations are now drawn. The profiles in the file are rotated 90 deg before extrusion, but this doesn't seem to have any effect (could be that it just happened to 'work' before and that the current behaviour is correct):

![Image](https://github.com/user-attachments/assets/d35fe259-094a-4aba-ba6a-ac393cbf0202)
